src/Database/Schema/Table.php

```diff
@@ -459,7 +459,9 @@ public function addIndex($name, $attrs)
             throw new Exception(sprintf('Index "%s" in table "%s" must have at least one column.', $name, $this->_table));
         }
         $attrs['columns'] = (array)$attrs['columns'];
-        foreach ($attrs['columns'] as $field) {
+        foreach ($attrs['columns'] as $parts) {
+            $parts = explode(' ', $parts);
+            $field = $parts[0];
```

Comment:
Firstly I had this approach (see my first commit) but then @lorenzo commented that it is 5.5 syntax, therefore I updated this. I'm not sure what is the correct version